addons/hr/models/hr_employee.py

```diff
@@ -1989,16 +1987,7 @@ def action_open_versions(self):
         }
 
     def _store_avatar_card_fields(self, res: Store.FieldList):
-        res.attr("company_id")
-        res.one("department_id", ["name"])
-        res.attr("work_email")
-        res.one("work_location_id", ["location_type", "name"])
-        res.attr("work_phone")
-        if res.target_user().has_group("hr.group_hr_user"):
-            # job_title is not a field of hr.employee.public, but it is a field of hr.employee
-            res.attr("job_title")
-        # HACK: fetch the employee fields from employees to retrieve hr.employee.public fields if no access to hr.employee
-        self.fetch(map(Store.get_field_name, res))
+        self.env["hr.employee.public"]._store_avatar_card_fields(res)
```

Comment:
With the new implementation this is not the correct way.

Method returning fields should be called on the correct recordset.

Fields of one model should not be substituted by a method on another model.

Either they are different but have the same fields, and the list should be duplicated, or one should use the other and then it should return the proper relation between them.

(`self` inside the method must be the recordset for which we want to return the fields, it's not supposed to be called on the model itself)